src/uu/seq/src/seq.rs

```diff
@@ -329,17 +206,85 @@ pub fn uu_app() -> App<'static, 'static> {
         )
 }
 
-fn done_printing<T: Num + PartialOrd>(next: &T, increment: &T, last: &T) -> bool {
+fn done_printing<T: Zero + PartialOrd>(next: &T, increment: &T, last: &T) -> bool {
     if increment >= &T::zero() {
         next > last
     } else {
         next < last
     }
 }
 
+/// Write a big decimal formatted according to the given parameters.
+///
+/// This method is an adapter to support displaying negative zero on
+/// Rust versions earlier than 1.53.0. After that version, we should be
+/// able to display negative zero using the default formatting provided
+/// by `-0.0f32`, for example.
+fn write_value_float(
+    writer: &mut impl Write,
+    value: &ExtendedBigDecimal,
+    width: usize,
+    precision: usize,
+    is_first_iteration: bool,
+) -> std::io::Result<()> {
+    let value_as_str = {
+        let s = if *value == ExtendedBigDecimal::MinusZero && is_first_iteration {
+            format!(
+                "-{value:>0width$.precision$}",
+                value = value,
+                width = if width > 0 { width - 1 } else { width },
+                precision = precision,
+            )
+        } else {
+            format!(
+                "{value:>0width$.precision$}",
+                value = value,
+                width = width,
+                precision = precision,
+            )
+        };
+        if *value == ExtendedBigDecimal::MinusZero && !s.starts_with('-') {
+            [String::from("-"), s].concat()
+        } else {
+            s
+        }
+    };
+    write!(writer, "{}", value_as_str)
+}
+
+/// Write a big int formatted according to the given parameters.
+fn write_value_int(
+    writer: &mut impl Write,
+    value: &ExtendedBigInt,
+    width: usize,
+    pad: bool,
+    is_first_iteration: bool,
+) -> std::io::Result<()> {
+    let value_as_str = if pad {
+        let s = if *value == ExtendedBigInt::MinusZero && is_first_iteration {
+            format!("-{value:>0width$}", value = value, width = width - 1,)
+        } else {
+            format!("{value:>0width$}", value = value, width = width,)
+        };
+        if *value == ExtendedBigInt::MinusZero && !s.starts_with('-') {
+            [String::from("-"), s].concat()
+        } else {
+            s
+        }
+    } else if *value == ExtendedBigInt::MinusZero && is_first_iteration {
+        format!("-{}", value)
+    } else {
+        format!("{}", value)
+    };
+    write!(writer, "{}", value_as_str)
+}
+
+// TODO `print_seq()` and `print_seq_integers()` are nearly identical,
```

Comment:
Maybe we could create a trait that both `ExtendedBigInt` and `ExtendedBigDecimal` implement and make `write_value`generic,  but we can do this at a later point.

Reply:
Yes definitely. If this pull request gets merged, I propose we open a new issue for that change. This diff is already larger than I usually like to produce.